src/core/bittorrent/torrenthandle.cpp

```diff
@@ -828,16 +838,29 @@ qulonglong TorrentHandle::eta() const
     const SpeedSampleAvg speed_average = m_speedMonitor.average();
 
     if (isSeed()) {
-        if (speed_average.upload == 0) return MAX_ETA;
-
         qreal max_ratio = maxRatio();
-        if (max_ratio < 0) return MAX_ETA;
+        int max_seeding_time = maxSeedingTime();
```

Comment:
Ok, will do, but only half of the original code is camelCased.  Just the C++ source code right? JavaScript will be a mess if it's camelCased.